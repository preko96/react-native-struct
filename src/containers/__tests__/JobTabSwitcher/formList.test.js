import React from 'react'
import { FormList } from '../../JobsTabSwitcher/formList'

jest.mock('react-native-sqlite-storage', () => {
	const mockSQLite = {
		openDatabase: () => {
			return {
				transaction: () => {
					() => { return [] }
				}
			}
		}
	}
	return mockSQLite
})

let wrapper
const mockLayout = {
	default: {
		CONTENT: { height: 1000, width: 1000 },
		MAP: { height: 300, width: 300 },
	},
	dynamic: {
		SEGMENT: { height: 100, width: 100 }
	}
}

beforeEach(()=> {
	wrapper = shallow(<FormList layoutHelper={mockLayout}/>)
})

it('renders without crashing', () => {
	expect(wrapper).toMatchSnapshot()
})

it('onLayout function setState: height properly', () => {
	wrapper.instance().onLayout()
	expect(wrapper.state('height')).toEqual(600)
})

it('calls GetJobInfo with the jobID', async ()=>{
	expect.assertions(1)
	const wrapper = mount(<FormList/>)
	expect(wrapper.state('forms')).toEqual([])
})



