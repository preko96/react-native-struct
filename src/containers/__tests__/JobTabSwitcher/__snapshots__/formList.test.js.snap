// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(List)
  dataArray={Array []}
  onLayout={[Function]}
  renderRow={[Function]}
  style={
    Object {
      "height": 98,
    }
  }
/>
`;
